jms/app/src/main/java/io/quarkus/ts/openshift/jms/PriceProducer.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.ts.openshift.jms;
+
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSContext;
+import javax.jms.Session;
+import java.util.Random;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A bean producing random prices every 5 seconds and sending them to the prices JMS queue.
+ */
+@ApplicationScoped
+public class PriceProducer implements Runnable {
+
+    @Inject
+    ConnectionFactory connectionFactory;
+
+    private final Random random = new Random();
+    private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    void onStart(@Observes StartupEvent ev) {
```

Comment:
I think this should really use Quarkus `@Scheduled`.

(I'm not sure how I'd improve the consumer, but I hope it can be done in a better way too.)

Reply:
+1

Reply:
@rsvoboda  @Ladicek   Well, we should make up our minds I guess :smiley: 

https://github.com/quarkusio/quarkus-quickstarts/blob/master/jms-quickstart/src/main/java/org/acme/jms/PriceProducer.java

Reply:
`@Scheduled` is nice to have atm, we can proceed without it, just using StartupEvent